lib/puppetserver/ca/action/generate.rb

```diff
@@ -77,31 +77,30 @@ def run(input)
           end
 
           # Load, resolve, and validate puppet config settings
-          puppet = Config::Puppet.parse(config_path)
+          settings_overrides = {}
+          settings_overrides[:certname] = input['certname'] unless input['certname'].empty?
+          settings_overrides[:ca_name] = input['ca_name'] unless input['ca_name'].empty?
+          settings_overrides[:subject_alt_names] = input['subject_alt_names'] unless input['subject_alt_names'].empty?
+          puppet = Config::Puppet.parse(config_path: config_path, cli_overrides: settings_overrides)
```

Comment:
Open to suggestions on better ways to do this. However, it does achieve the desired results of getting these overrides in before interpolation.